packages/core/src/modules/dids/methods/peer/peerDidNumAlgo2.ts

```diff
@@ -69,10 +69,13 @@ export function didToNumAlgo2DidDocument(did: string) {
       for (let service of services) {
         // Expand abbreviations used for service key/values
         service = expandServiceAbbreviations(service)
-
         service.id = `${did}#${service.type.toLowerCase()}-${serviceIndex++}`
-
-        didDocument.addService(JsonTransformer.fromJSON(service, DidDocumentService))
+        try {
+          didDocument.addService(JsonTransformer.fromJSON(service, DidDocumentService))
+        } catch (e) {
+          //Ignore a service if we do not recognize it
+          serviceIndex--
```

Comment:
We should nog skip the service index i think, as then the index references will be messed up. I think we can skip adding it to the document, but also skip the index

Reply:
Also furious when does this happen? If a json transform fails? I don't think we should just swallow such an error. Can you explain why this is needed?

Reply:
Looking at this I realize this was a bit of a hold over from implementing this functionality in another library that had much stricter serialization rules than typescript does, specifically with unknown additional keys appearing causing errors. With the regex adjustment this is redundant for Credo.

Reply:
Removed this logic as it is not needed in Credo because serialization is more fault tolerant than I thought at development time.

Reply:
The code is still here?

Reply:
With my comment below about did:peer:2 adding multiple services this was added because some of the agents and mediator I tested against included didCommV2 services that were intended to be ignored by the agent if they do not use didCommV2.  Have this try catch here successfully ignores it and keeps the indexes of the services we do recognize consistent.

Reply:
But by doing this the references between implementations will mess up. We would skip didcomm v2 and others would not skip it and thus the same id could point to different services. I think we should not do this

Reply:
AFter looking at this in more depth -- we don't handle the DIDComm v2 serviceEndpoint being an object. So instead of doing this, we need to fix that parsing of serviceEndpoint instead

Reply:
Do you have example did:peer:2 document with DIDComm v2 service for test? I have a PR locally that correctly fixes https://github.com/openwallet-foundation/credo-ts/issues/1789, and thus should also fix this issue